index.js

```diff
@@ -77,38 +77,72 @@ class GuideBot extends Client {
-        props.init(this);
+      switch (type) {
+        case "command": {
+          const props = new (require(modulePath))(this);
+          props.conf.location = modulePath;
+          if (props.init) {
+            props.init(this);
+          }
+      
+          this.commands.set(props.help.name, props);
+          props.conf.aliases.forEach(alias => {
+            this.aliases.set(alias, props.help.name);
+          });
+          return false;
+        }
+  
+        case "slash": {
+          const props = new (require(modulePath))(this);
+          this.logger.log(`Loading slash command: ${moduleName}. 👌`, "log");
+          props.conf.location = modulePath;
+          client.slashcmds.set(props.commandData.name, props);
+          break;
+        }
+  
+        case "event": {
+          client.logger.log(`Loading event: ${moduleName}. 👌`, "log");
+          const event = new(require(modulePath))(this);
+          client.events.set(event.conf.name, event);
+  
+          // This line is awesome by the way. Just sayin'.
+          client.on(moduleName, (...args) => event.run(...args));
+          delete require.cache[require.resolve(modulePath)];
+        }
       }
-
-      this.commands.set(props.help.name, props);
-      props.conf.aliases.forEach(alias => {
-        this.aliases.set(alias, props.help.name);
-      });
-      return false;
     } catch (e) {
-      return `Unable to load command ${commandName}: ${e}`;
+      return `Unable to load ${type} ${moduleName}: ${e}`
     }
   }
 
-  async unloadCommand(commandPath, commandName) {
-    let command;
-    if (this.commands.has(commandName)) {
-      command = this.commands.get(commandName);
-    } else if (this.aliases.has(commandName)) {
-      command = this.commands.get(this.aliases.get(commandName));
-    }
-    if (!command) return `The command \`${commandName}\` doesn't seem to exist, nor is it an alias. Try again!`;
+  async unloadModule(type, commandPath, commandName) {
+    switch (type) {
+      case "command": {
+        let command;
+        if (this.commands.has(commandName)) {
+          command = this.commands.get(commandName);
+        } else if (this.aliases.has(commandName)) {
+          command = this.commands.get(this.aliases.get(commandName));
+        }
+        if (!command) return `The command \`${commandName}\` doesn't seem to exist, nor is it an alias. Try again!`;
+
+        if (command.shutdown) {
+          await command.shutdown(this);
+        }
+        delete require.cache[require.resolve(commandPath)];
+        return false;
+      }
 
-    if (command.shutdown) {
-      await command.shutdown(this);
+      case "slash": {
+        let command;
+        if (this.slashcmds.has(commandName)) command = this.slashcmds.get(commandName);
+        if (!command) return `The command \`${commandName}\` doesn't seem to exist, nor is it an alias. Try again!`;
+        delete require.cache[require.resolve(`${commandPath}${path.sep}${commandName}.js`)];
```

Comment:
This line needs to be `commandPath` only otherwise it will throw the following error when you're trying to reload a slash command;
```
Cannot find module 'C:\Users\YorkAARGH\Desktop\guidebot-class-v13-update\slash\miscellaneous\ping.js\ping.js'
Require stack:
- C:\Users\YorkAARGH\Desktop\guidebot-class-v13-update\index.js
```

Reply:
Fixed.